apps/laboratory/tests/shared/utils/email.ts

```diff
@@ -77,10 +77,17 @@ export class Email {
     throw new Error(`No code found in email: ${body}`)
   }
 
-  getEmailAddressToUse(index: number, domain = EMAIL_DOMAIN): string {
-    const randIndex = Math.floor(Math.random() * 10) % 9
+  async getEmailAddressToUse(domain = EMAIL_DOMAIN): Promise<string> {
+    const response = await fetch(
+      'https://id-allocation-service.walletconnect-v1-bridge.workers.dev/allocate'
+    )
+    const { id } = await response.json()
 
-    return `w3m-w${index}${randIndex}@${domain}`
+    const email = `w3m-w${id}@${domain}`
+    // eslint-disable-next-line no-console
+    console.log(`allocating email: ${email}`)
```

Comment:
Logging this for now so we can detect race conditions

Reply:
Race conditions should be impossible now, but still logging just in-case for now. We can remove in a later PR